core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseSampleSourceTelemetry.java

```diff
@@ -17,6 +17,8 @@ public void reset() {
     @Override
     protected void setSampleRate(Envelope envelope) {
         Double currentSP = getSamplingPercentage();
-        envelope.setSampleRate(currentSP);
+        if (currentSP != null) {
```

Comment:
so how the `reset` worked before? It calls this method and passes `null`. Perhaps it should be a fix in serialization and null is a `valid` value.

Reply:
@SergeyKanzhelev upon closely looking at the code I found following flow:
1. setSampleRate() and setSamplePercentage() are two methods.
2. User has the permission to set sampling percentage which in-turn is internally set to sampling rate. 
3. when the telemetry.reset() is called it sets the sampling rate as null and next time when the telemtry is pushed, it sees that sampling rate is null and picks the default sampling rate as 100.0D

Therefore, I believe there should be no need of change though I might be wrong. Please let me know what are your thoughts.